spec/spec_helper.rb

```diff
@@ -19,6 +18,11 @@ def fixture_file path
     File.join fixtures_dir, path
   end
 
+  def testname
+    metadata = method_missing(:class).metadata
```

Comment:
This could be stored using the following:

```rb
config.before :each do |example|
  @example_group_name = File.basename (example.instance_variable_get :@metadata)[:file_path], '.rb'
end
```

Reply:
In fact, it's even simpler:

```rb
RSpec.configure do |config|
  config.expose_current_running_example_as :example

  config.before :each do
    @example_group_name = File.basename example.file_path, '.rb'
  end
end
```

Reply:
I have removed the testname-based structure in the fixtures section as it's currently just one.
There's also the need to point to the proper path in case the test fails, and at that time, the :example doesn't contain the name yet or not anymore.